### ISSUE TITLE
CodeChecker check no longer works

### ISSUE THREAD

**Issue by technic** [Ericsson/codechecker#2505]
**Describe the bug**
Since few releases ago I am getting an error when I try to check my project
```
$ CodeChecker check -b "make" -o ./reports
 No handlers could be found for logger "system"
 Traceback (most recent call last):
   File "/root/codechecker/build/CodeChecker/cc_bin/CodeChecker.py", line 111, in main
     subcommands[subcommand])
   File "/root/codechecker/build/CodeChecker/cc_bin/CodeChecker.py", line 47, in add_subcommand
     command_module.add_arguments_to_parser(sc_parser)
   File "/root/codechecker/build/CodeChecker/cc_bin/../lib/python2.7/codechecker_analyzer/cmd/check.py", line 321, in add_arguments_to_parser
     clang_has_z3 = analyzer_types.is_z3_capable(context)
   File "/root/codechecker/build/CodeChecker/lib/python2.7/codechecker_analyzer/analyzers/analyzer_types.py", line 70, in is_z3_capable
     analyzer_env)
   File "/root/codechecker/build/CodeChecker/lib/python2.7/codechecker_analyzer/host_check.py", line 83, in has_analyzer_option
     LOG.debug('run: "%s"', ' '.join(cmd))
 TypeError: sequence item 0: expected string, NoneType found
```

**CodeChecker version**
Built from Git tags: v6.11.0 (v6.11.0)

**To Reproduce**
Full log of the GitLab CI job https://gitlab.com/technic93/e2designer/-/jobs/387534282

**Expected behavior**
Should work as before

**Desktop (please complete the following information)**
 - OS: Linux
 - Browser: irrelevant

**Additional context**
Did something in the CodeChecker comandline interface changed? Thanks.

**Comment by csordasmarton** [comment]
@technic I think you are using only Clang Tidy, and Clang SA is not available on you machine. We do not handle this use case so I created a simple patch which will fix this: #2506.

**Comment by technic** [comment]
thanks for your reply. But what is clang SA and how to install it? I have `scan-build` tool and `clang-tidy` tool in the $PATH is there something else?

**Comment by whisperity** [comment]
ClangSA is the [Clang Static Analyzer](http://clang-analyzer.llvm.org). It's not a separate binary (for various reasons), so if you have `clang`, the compiler, installed on your system, it contains Clang SA.

**Comment by whisperity** [comment]
I think you could alternatively specify `--analyzers clang-tidy` to check and it'll skip trying to look up ClangSA.

**Comment by technic** [comment]
Hi, I see a have `clang` the compiler in my docker image but I did not symlink it from `clang-8` to `clang`, I only did it for `clang++` because my project is in C++. Now I provide both `clang` and `clang++` in $PATH and it works fine.

Basically the issue is solved, but I think it would be better to provide more meaningful error in this case.

**Comment by gyorb** [comment]
Yes, you are right thanks for reporting the issue.
The problem was pretty early at the argument checking part were some clang static analyzer features are tested even it it was not available. We have plans to change this behavior to show all the arguments and only warn the user if he tries to use it and it is not available: #2432
